workspaces/tech-insights/plugins/tech-insights/src/api/TechInsightsClient.ts

```diff
@@ -106,10 +107,20 @@ export class TechInsightsClient implements TechInsightsApi {
     });
   }
 
+  private getCacheKey(path: string, init?: RequestInit): string {
```

Comment:
maybe we could use https://www.npmjs.com/package/swr for fetching?

Reply:
The problem is that it is only a hook, and this logic is within an API. Perhaps we could build a simplistic version of caching fetch requests, some kind of `cachedFetchApi` (using `fetchApi`) which allows for concurrent (or somewhat concurrent) exact same requests to reuse the same call?